src/jwt/macros.rs

```diff
@@ -40,3 +40,101 @@ macro_rules! check_symmetric_key {
         key
     }};
 }
+
+#[macro_export]
+macro_rules! check_public_key {
```

Comment:
I think `check_public_key` and `check_private_key` can be merged into one macro. smth like that:
```Rust
macro_rules! check_rsa_key {
    (public_key: $public_key:expr, name: $name:expr, notificator: $notificator:expr, key: $key:ty) => {{
        // ....
    }};
}
```

and then:
```Rust
// ...
let public_key = check_rsa_key!(
    public_key: key,
    name: jwt.signature_algorithm.to_string(),
    notificator: spawn_notification,
    PublicKey
);
// or
let private_key = check_private_key!(
    private_key: key,
    name: jwt.signature_algorithm.to_string(),
    notificator: &spawn_notification,
    PrivateKey
);
```